### ISSUE TITLE
Floating point exception

### ISSUE THREAD

**Issue by strongcourage** [pts/sam2p#66]
Hi,

I found a FPE bug on the latest commit `cafd4b8` of master.

PoC: https://github.com/strongcourage/PoCs/blob/master/sam2p_cafd4b8/PoC_fpe
Command: sam2p $PoC /tmp/out.bmp

ASAN says:
~~~
This is sam2p 0.49.4.
Available Loaders: PS PDF JAI PNG JPEG TIFF PNM BMP GIF LBM XPM PCX TGA.
Available Appliers: XWD Meta Empty BMP PNG TIFF6 TIFF6-JAI JPEG-JAI JPEG PNM GIF89a+LZW XPM PSL1C PSL23+PDF PSL2+PDF-JAI P-TrOpBb.
sam2p-asan: Warning: TGA: 24 bit image, 9 bit alpha is greater than 24 total bits per pixel
sam2p-asan: Warning: TGA: reducing to * bit alpha: 0
Floating point exception
~~~
Thanks,
Manh Dung

**Comment by pts** [comment]
Thank you for reporting this! Fixed in b953f63307c4a83fa4615a4863e3fb250205cd98.

The new behavior is failing with a more informative error message:

```
sam2p: Error: Image: Image of size 0.
```

**Comment by fgeek** [comment]
CVE-2020-19492 has been assigned for this issue.
